kore/src/Kore/Strategies/Verification.hs

```diff
@@ -363,10 +365,24 @@ transitionRule' claims axioms =
     & withConfiguration
     & withDebugProofState
     & logTransitionRule
+    & countExecutionDepth
   where
     axiomGroups = groupSortOn Attribute.Axiom.getPriorityOfAxiom axioms
 
-
+countExecutionDepth
+    :: TransitionRule m ReachabilityRule
+    -> TransitionRule m ReachabilityRule
+countExecutionDepth rule = \prim proofState -> do
+    result <- rule prim proofState
+    if prim `elem` [Prim.ApplyClaims, Prim.ApplyAxioms]
+        && isGoalRemainder result
+    then
+        fmap incrementDepth (rule prim proofState)
+    else
+        rule prim proofState
```

Comment:
This would call `rule` twice. It must return the `result` instead of calling `rule` again.